comp/core/config/config_mock.go

```diff
@@ -37,41 +37,34 @@ func (m mockDependencies) getSecretResolver() (secrets.Component, bool) {
 // newMock exported mock builder to allow modifying mocks that might be
 // supplied in tests and used for dep injection.
 func newMock(deps mockDependencies, t testing.TB) (Component, error) {
-	backupConfig := pkgconfigmodel.NewConfig("", "", strings.NewReplacer())
-	backupConfig.CopyConfig(pkgconfigsetup.Datadog())
-
-	pkgconfigsetup.Datadog().CopyConfig(pkgconfigmodel.NewConfig("mock", "XXXX", strings.NewReplacer()))
-
-	env.SetFeatures(t, deps.Params.Features...)
-
-	// call InitConfig to set defaults.
-	pkgconfigsetup.InitConfig(pkgconfigsetup.Datadog())
-	c := &cfg{
-		Config: pkgconfigsetup.Datadog(),
-	}
+	var mockConf model.Config
 
-	if !deps.Params.SetupConfig {
-		if deps.Params.ConfFilePath != "" {
-			pkgconfigsetup.Datadog().SetConfigType("yaml")
-			err := pkgconfigsetup.Datadog().ReadConfig(strings.NewReader(deps.Params.ConfFilePath))
-			if err != nil {
-				// The YAML was invalid, fail initialization of the mock config.
-				return nil, err
-			}
-		}
+	if deps.Params.ConfFilePath != "" {
+		mockConf = mock.NewFromFile(t, deps.Params.ConfFilePath)
 	} else {
-		warnings, _ := setupConfig(pkgconfigsetup.Datadog(), deps)
-		c.warnings = warnings
+		mockConf = mock.New(t)
 	}
 
-	// Overrides are explicit and will take precedence over any other
-	// setting
+	// Overrides are explicit and will take precedence over any other setting
 	for k, v := range deps.Params.Overrides {
-		pkgconfigsetup.Datadog().SetWithoutSource(k, v)
+		mockConf.SetWithoutSource(k, v)
 	}
 
-	// swap the existing config back at the end of the test.
-	t.Cleanup(func() { pkgconfigsetup.Datadog().CopyConfig(backupConfig) })
+	setup.LoadProxyFromEnv(mockConf)
```

Comment:
I assume we need to call `LoadProxyFromEnv` because we are no longer calling `pkgconfigsetup.InitConfig(pkgconfigsetup.Datadog())`, is that correct?

Reply:
Yes